src/shogun/base/SGObject.h

```diff
@@ -124,6 +124,12 @@ class CSGObject
 	/** destructor */
 	virtual ~CSGObject();
 
+	template<class Archive>
```

Comment:
it's the user's choice... we can create helper functions if we want, say `saveJSON`, `saveXML` etc. but the idea here is that the user could define it's own Archive type as well.